gui/wxpython/main_window/frame.py

```diff
@@ -912,6 +917,24 @@ def OnCBPageClosing(self, event):
 
         event.Skip()
 
+    def OnMapNotebookPageChanged(self, event):
+        """Page in map notebook changed.
+        Also change active tab in Display panel."""
+        if self.workspace_manager.loadingWorkspace:
+            return
+
+        try:
+            currentMapDisp = self.mapnotebook.GetCurrentPage()
+            pgnum = currentMapDisp.getLayerTreeIndex()
+            if pgnum > -1:
+                self.notebookLayers.SetSelection(pgnum)
+                self.currentPage = self.notebookLayers.GetCurrentPage()
+                self.currentPageNum = self.notebookLayers.GetSelection()
+        except Exception:
```

Comment:
What exception is actually thrown here? It's better to use the specific one to make it more clear what it is trying to do.

Reply:
It is TypeError. As I understand it right, the OnMapNotebookChanged is also called during the processing of NewDisplay method. The map panel is already known but the layertree not, so thats the reason why it throws this exception.